gtsam/hybrid/tests/testIncrementalHybrid.cpp

```diff
@@ -45,296 +44,334 @@ using symbol_shorthand::X;
+   * factor 3:  [x4 | m3 m2 m1 ], 8 components
+   */
+  auto hybridBayesNet = incrementalHybrid.hybridBayesNet();
+
+  EXPECT_LONGS_EQUAL(4, hybridBayesNet.size());
+  EXPECT_LONGS_EQUAL(2, hybridBayesNet.atGaussian(0)->nrComponents());
+  EXPECT_LONGS_EQUAL(4, hybridBayesNet.atGaussian(1)->nrComponents());
+  EXPECT_LONGS_EQUAL(5, hybridBayesNet.atGaussian(2)->nrComponents());
+  EXPECT_LONGS_EQUAL(5, hybridBayesNet.atGaussian(3)->nrComponents());
+
+  auto &lastDensity = *(hybridBayesNet.atGaussian(3));
+  auto &unprunedLastDensity = *(unprunedHybridBayesNet->atGaussian(3));
+  std::vector<std::pair<DiscreteValues, double>> assignments =
+      discreteFactor_m1.enumerate();
+  // Loop over all assignments and check the pruned components
+  for (auto &&av : assignments) {
+    const DiscreteValues &assignment = av.first;
+    const double value = av.second;
+
+    if (value == 0.0) {
+      EXPECT(lastDensity(assignment) == nullptr);
+    } else {
+      CHECK(lastDensity(assignment));
+      EXPECT(assert_equal(*unprunedLastDensity(assignment),
+                          *lastDensity(assignment)));
+    }
+  }
+}
+
+/* ****************************************************************************/
+// Test approximate inference with an additional pruning step.
+TEST(DCGaussianElimination, Incremental_approximate) {
+  Switching switching(5);
+  IncrementalHybrid incrementalHybrid;
+  GaussianHybridFactorGraph graph1;
+
+  // Add the 3 DC factors, x1-x2, x2-x3, x3-x4
+  for (size_t i = 0; i < 3; i++) {
+    graph1.push_back(switching.linearizedFactorGraph.dcGraph().at(i));
+  }
+
+  // Add the Gaussian factors, 1 prior on X(1), 4 measurements
+  for (size_t i = 0; i <= 4; i++) {
+    graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(i));
+  }
+
+  // Create ordering.
+  Ordering ordering;
+  for (size_t j = 1; j <= 4; j++) {
+    ordering += X(j);
+  }
+
+  // Run update with pruning
+  size_t maxComponents = 5;
+  incrementalHybrid.update(graph1, ordering, maxComponents);
+
+  // Test the bayes net and if the number of pruned leaves are as expected.
+  auto actualBayesNet1 = incrementalHybrid.hybridBayesNet();
+  CHECK_EQUAL(4, actualBayesNet1.size());
+  EXPECT_LONGS_EQUAL(2, actualBayesNet1.atGaussian(0)->nrComponents());
+  EXPECT_LONGS_EQUAL(4, actualBayesNet1.atGaussian(1)->nrComponents());
+  EXPECT_LONGS_EQUAL(8, actualBayesNet1.atGaussian(2)->nrComponents());
+  EXPECT_LONGS_EQUAL(5, actualBayesNet1.atGaussian(3)->nrComponents());
+
+  /***** Run Round 2 *****/
+  GaussianHybridFactorGraph graph2;
+  graph2.push_back(switching.linearizedFactorGraph.dcGraph().at(3));
+  graph2.push_back(switching.linearizedFactorGraph.gaussianGraph().at(5));
+
+  Ordering ordering2;
+  ordering2 += X(4);
+  ordering2 += X(5);
+
+  // Run update with pruning a second time.
+  incrementalHybrid.update(graph2, ordering2, maxComponents);
+
+  // Test the bayes net and if the number of pruned leaves
+  // are as expected after round 2.
```

Comment:
As above, the point of the comment is to say what you're expecting and why :-)

Reply:
I understand what you mean after seeing your current comments. Going to do a full update. :)